### PR TITLE
add golden-output regression test for PatchTSTForecaster

### PR DESCRIPTION
**Pull request by Nischal1425**
#### Reference Issues/PRs

Part of #10672, adds golden-output regression tests for `PatchTSTForecaster`.

Stacks over #10940.

#### What does this implement/fix? Explain your changes.

Adds a regression test validating `PatchTSTForecaster` output against the original implementation, by calling `transformers.PatchTSTForPrediction` directly and bypassing sktime entirely to produce the reference values.

Reference outputs were generated with:

```python
import numpy as np, torch
from transformers import PatchTSTForPrediction

model = PatchTSTForPrediction.from_pretrained(
    "ibm-granite/granite-timeseries-patchtst"
)
model.eval()
y = np.random.RandomState(0).randn(512, 7).astype("float32")
with torch.no_grad():
    out = model(past_values=torch.from_numpy(y).unsqueeze(0))
out.prediction_outputs[0].numpy()[:3, :3]
```

sktime's wrapper matches this **exactly** comparing the full output grid, maximum absolute difference is `0.0`, i.e. bit-identical rather than merely within tolerance. The stored slice is kept small (first 3 steps × 3 channels) per the issue guidance, with the generating script embedded as a comment in the test file for reproducibility.

Two deviations from the issue's defaults, both explicitly permitted by it:

- **`load_airline` is not usable here.** This checkpoint is trained on ETTh1 and requires `num_input_channels=7` and `context_length=512`; airline is univariate with 144 points. A fixed-seed `RandomState(0)` array is used instead ("random data is acceptable if more suitable").
- **Checkpoint choice.** The one in the estimator's docstrings (`namctin/patchtst_etth1_forecast`) is now gated and 403s, so it cannot be used in CI. `ibm-granite/granite-timeseries-patchtst` is public, widely used, and loads with zero missing or mismatched keys. Note that `ibm-research/patchtst-etth1-pretrain` is *not* a viable alternative, it is a masked-pretraining checkpoint with no forecast head.

#### Does your contribution introduce a new dependency? If yes, which one?

No, `transformers`, `torch` and `accelerate` are already this estimator's declared `python_dependencies`.

#### What should a reviewer concentrate their feedback on?

Whether `ibm-granite/granite-timeseries-patchtst` is the right checkpoint to pin the golden values to, given the docstring checkpoint is no longer accessible.

#### Did you add any tests for the change?

Yes, this PR is itself the test, `test_patch_tst_predictions_match_transformers_reference`, gated with
`run_test_for_class` per the issue. Also updated the `tests:specific` tag on `PatchTSTForecaster` to include the new test file.

The horizon is `fh=1..24`, deliberately below the checkpoint's native `prediction_length` of 96, so this test is independent of #10940 and passes on unmodified `main`.

#### Any other comments?

I hit a genuine bug in this estimator while generating these references an off-by-one making `fh` at the checkpoint's native `prediction_length` fail under `fit_strategy="zero-shot"`. That is filed separately as #10940 to keep the bug fix reviewable apart from test coverage.

**Merge note:** this PR and #10940  both create `sktime/forecasting/tests/test_patch_tst.py` and both add the same `tests:specific` tag line, so they will conflict. Whichever merges second needs a trivial rebase, happy to do that, or to stack them, whichever you prefer.

### CODE REVIEW

**Review by geetu040 — APPROVED** [review]
Looks good, waiting for the CI to be green.

Stacks over #10940
